ProductsService/ProductsService.API/Controllers/ProductController.cs

```diff
@@ -0,0 +1,168 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using ProductsService.Application.Common.Dto.Requests;
+using ProductsService.Application.UseCases.ProductUseCases.Commands.Create;
+using ProductsService.Application.UseCases.ProductUseCases.Commands.CreateCategory;
+using ProductsService.Application.UseCases.ProductUseCases.Commands.CreateImage;
+using ProductsService.Application.UseCases.ProductUseCases.Commands.Delete;
+using ProductsService.Application.UseCases.ProductUseCases.Commands.DeleteCategory;
+using ProductsService.Application.UseCases.ProductUseCases.Commands.DeleteImage;
+using ProductsService.Application.UseCases.ProductUseCases.Commands.Update;
+using ProductsService.Application.UseCases.ProductUseCases.Queries.GetByFilters;
+using ProductsService.Application.UseCases.ProductUseCases.Queries.GetById;
+using ProductsService.Application.UseCases.ProductUseCases.Queries.GetByUserId;
+
+namespace ProductsService.API.Controllers
+{
+    [Route("api/products")]
+    [ApiController]
+    public class ProductController(
+        IMediator mediator) : ControllerBase
+    {
+        [HttpGet]
+        public async Task<IActionResult> GetAll([FromQuery] GetProductsByFiltersRequest request, CancellationToken cancellationToken)
+        {
+            var products = await mediator.Send(request, cancellationToken);
+
+            return Ok(products);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetById(Guid id, CancellationToken cancellationToken)
+        {
+            var request = new GetProductByIdRequest(id);
+
+            var product = await mediator.Send(request, cancellationToken);
+
+            return Ok(product);
+        }
+
+        [HttpGet("user/{id}")]
+        public async Task<IActionResult> GetAllByUserId(Guid id, [FromQuery] FiltersRequestDto dto, CancellationToken cancellationToken)
+        {
+            var request = new GetProductsByUserIdRequest(
+                id,
+                dto.Category.ToString(),
+                dto.Name,
+                dto.OrderBy, 
+                dto.Asc,
+                dto.Page,
+                dto.PageSize);
+
+            var product = await mediator.Send(request, cancellationToken);
+
+            return Ok(product);
+        }
+
+        [Authorize]
+        [HttpPost]
+        public async Task<IActionResult> Create([FromForm] CreateProductRequestDto dto, CancellationToken cancellationToken)
+        {
+            var userId = Guid.Parse(User.FindFirst("userId")!.Value);
+
+            var command = new CreateProductCommand(
```

Comment:
the same

Reply:
I used record for dtos, and it cant be updated after initialization. But i can switch on class